brightener.h

```diff
@@ -1,17 +1,35 @@
 #pragma once
 #include <cstdint>
+#include <stdexcept>
+#include <memory> 
+constexpr int MAX_ROWS = 1024;
+constexpr int MAX_COLS = 1024;
 
 struct Image {
-	int rows;
-	int columns;
-	uint8_t pixels[1024 * 1024]; // max 1k x 1k image
-};
+    int rows;
+    int columns;
+    std::unique_ptr<uint8_t[]> pixels;
+
+    Image(int r, int c) : rows(r), columns(c), pixels(nullptr) {
+        if (rows > MAX_ROWS || columns > MAX_COLS) {
+            throw std::invalid_argument("Image dimensions exceed maximum allowed size (1024x1024).");
+        }
+        pixels = std::make_unique<uint8_t[]>(rows * columns);
+        std::fill_n(pixels.get(), rows * columns, 0);
+    }
+    // Disable copy constructor and copy assignment operator to prevent shallow copies
+    Image(const Image&) = delete;
+    Image& operator=(const Image&) = delete;
 
+    // Allow move constructor and move assignment operator
+    Image(Image&&) noexcept = default;
+    Image& operator=(Image&&) noexcept = default;
+};
 class ImageBrightener {
 private:
-	Image m_inputImage;
+	Image& m_inputImage;
```

Comment:
Having a reference data member often makes the lifetimes tricky. Here, the `ImageBrightener` object has a reference to `Image`. That `Image` object will then need to be around, at least as long as this `ImageBrightener` object. Why not use a `unique_ptr` or a `shared_ptr` instead?